compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala

```diff
@@ -287,18 +287,24 @@ object SyntaxHighlighting {
         case '@' => true
         case ',' => true
         case '.' => true
-        case ',' => true
-        case ')' => true
         case _ => false
       }
 
-      def shouldUpdateLastToken(currentPotentialToken: String): Boolean =
-        (lastToken, currentPotentialToken) match {
-          case (_, ("var" | "val" | "def" | "case")) => true
-          case (("val" | "var"), "=") => true
-          case ("case", ("=>" | "class" | "object")) => true
-          case ("def", _) => true
-          case _ => false
+      val valDefStarterTokens = "var" :: "val" :: "def" :: "case" :: Nil
+
+      /** lastToken is used to check whether we want to show something
```

Comment:
Maybe we shouldn't call it lastToken then since it's not the last token anymore :).